src/log_format_impls.cc

```diff
@@ -146,7 +146,7 @@ class generic_log_format : public log_format {
                 &level)) != NULL) {
             const char *level_str = &sbr.get_data()[level.c_begin];
             logline::level_t level_val = logline::string2level(
-                    level_str, level.length());
+                    level_str, strlen(level_str));
```

Comment:
level.length() would be something that I want if the line actually matched a known level. However, that's not what I want if it doesn't match a known level and needs to be matched against LEVEL_RE regex. If you look at the implementation of string2level, if the level is unknown after a call to 'abbrev2level', it is matched against LEVEL_RE to search for log levels. However, that uses the incorrect level.length() to size the pcre input and ends up not identifying the levels if it isn't the first thing after the timestamp.

You can try this `echo "something something error" | lnav -t`. The log format is correctly identified as 'generic_log' but the LEVEL_RE does not match the levels. If you try `echo "error" | lnav -t`, that will correctly identify the log level. That was the problem I was trying to rectify but I guess, incorrectly. Or maybe that is by design.

Here are some more odd examples:

`echo "error something something" | lnav -t`  -> level not recognized
`echo "error:something something" | lnav -t` -> level recognized